packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue

```diff
@@ -70,19 +79,84 @@ export default defineComponent({
     }
   },
   setup() {
+    const store = useStore()
+    const { openEditor } = useFileActions()
+    const clientService = useClientService()
+    const { webdav } = clientService
     const { $gettext } = useGettext()
     const appIconKey = ref('')
+    const { getMatchingSpace } = useGetMatchingSpace()
 
     const applicationSwitcherLabel = computed(() => {
       return $gettext('Application Switcher')
     })
     const updateAppIcons = () => {
       appIconKey.value = uuid.v4().replaceAll('-', '')
     }
+    const currentFolder = computed((): SpaceResource => {
+      return store.getters['Files/currentFolder']
+    })
+    const files = computed((): Array<Resource> => store.getters['Files/files'])
+
+    const onEditorApplicationClick = async (item: any) => {
+      console.log(getMatchingSpace(unref(currentFolder)))
+      let destinationSpace = unref(currentFolder)
+      let destinationFiles = unref(files)
+
+      if (!destinationSpace || !destinationSpace.canCreate()) {
+        destinationSpace = unref(store.getters['runtime/spaces/spaces']).find(
```

Comment:
We will change to personal drive. 
for example this happens while clicking text editor being on admin settings route or on read only share